src/components/search-by-encounters/search-by-encounters.component.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from "react";
+
+import { showNotification } from "@openmrs/esm-framework";
+import {
+  DatePicker,
+  DatePickerInput,
+  Column,
+  NumberInput,
+  Dropdown,
+  MultiSelect,
+} from "carbon-components-react";
+import dayjs from "dayjs";
+import { useTranslation } from "react-i18next";
+
+import { fetchLocations } from "../../cohort-builder.resource";
+import { Form, SearchByProps, Location } from "../../types";
+import SearchButtonSet from "../search-button-set/search-button-set";
+import {
+  fetchEncounterTypes,
+  fetchForms,
+} from "./search-by-encounters.resources";
+import styles from "./search-by-encounters.style.scss";
+import { getDescription, getQueryDetails } from "./search-by-encounters.utils";
+
+const SearchByEncounters: React.FC<SearchByProps> = ({ onSubmit }) => {
+  const { t } = useTranslation();
+  const [atLeastCount, setAtLeastCount] = useState(0);
+  const [atMostCount, setAtMostCount] = useState(0);
+  const [encounters, setEncounters] = useState([]);
+  const [selectedEncounterTypes, setSelectedEncounterTypes] = useState([]);
+  const [encounterLocation, setEncounterLocation] = useState<Location>();
+  const [encounterForm, setEncounterForm] = useState<Form>();
+  const [locations, setLocations] = useState<Location[]>([]);
+  const [forms, setForms] = useState<Form[]>([]);
+  const [onOrBefore, setOnOrBefore] = useState("");
+  const [onOrAfter, setOnOrAfter] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
Instead of using numerous useStates, it would be better to either make a single useState, or useContext. Else there'll be multiple re-renders.

Reply:
Yes, I agree with @anjula-sack,  `useState` has a special optimization for atomic values. You can use objects for values that change together.